### PR TITLE
Unexpected ts migration for LLM

### PR DESCRIPTION
**Pull request by juan-cortes**
image

# This turned into an unexpected typescript migration and I don't know exactly how it happened.

### 📝 Description
I promise it was not my intention to hijack the purpose of https://github.com/LedgerHQ/ledger-live/pull/991 but it ended up being basically what this is. Following the proposed approach from @LFBarreto over there I went over all the files in the repo, and completed a migration. We have a bunch of house cleaning to do after this though, we need to go over all the warnings, all the unused files, etc. Left more data for this over on slack but it's not yet published so I can't link there just yet.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` 
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-2505` 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo
No demo needed.

### 🚀 Expectations to reach
Nothing to test if CI is happy.

### CODE REVIEW

**Review by hedi-edelbloute — APPROVED** [review]

**Review by ofreyssinet-ledger — APPROVED** [review]

**Comment by ofreyssinet-ledger** [comment]
you're not a pawn, you're the chess player, I'm just a spectator making comments hahaha

File: apps/ledger-live-mobile/.eslintrc.js [hunk 1]
```diff
@@ -13,7 +14,7 @@ module.exports = {
       },
     },
   },
-  plugins: ["prettier", "detox"],
+  plugins: ["prettier", "detox", "@typescript-eslint"],
```
Comment (Justkant) [inline apps/ledger-live-mobile/.eslintrc.js:17]: This should not be needed when using the recommended rules config: https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin#usage
```suggestion
  plugins: ["prettier", "detox"],
```

File: apps/ledger-live-mobile/.eslintrc.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ module.exports = {
     "airbnb",
     "prettier",
     "plugin:json/recommended",
+    "plugin:@typescript-eslint/recommended",
```
Comment (Justkant) [inline apps/ledger-live-mobile/.eslintrc.js:8]: Should we also add the typed rules ? https://typescript-eslint.io/docs/linting/typed-linting/

File: .changeset/eighty-hairs-nail.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"live-mobile": patch
+---
+
+Added ts eslint parser for LLM
```
Comment (Justkant) [inline .changeset/eighty-hairs-nail.md:5]: CMIIW but I don't think we need a changeset for that kind of changes

File: apps/ledger-live-mobile/package.json [hunk 4]
```diff
@@ -231,10 +231,12 @@
...
     "@types/react-redux": "^7.1.24",
     "@types/react-test-renderer": "^17.0.1",
-    "@types/styled-components": "^5.1.25",
-    "@types/styled-system": "^5.1.15",
     "@types/redux-actions": "^2.6.2",
     "@types/semver": "^7.3.9",
+    "@types/styled-components": "^5.1.25",
+    "@types/styled-system": "^5.1.15",
+    "@typescript-eslint/eslint-plugin": "^5.33.1",
+    "@typescript-eslint/parser": "^5.33.1",
```
Comment (Justkant) [inline apps/ledger-live-mobile/package.json:239]: No lockfile update from this addition ?

**Comment by juan-cortes** [comment]
Let's see how badly ci hates this

**Comment by juan-cortes** [comment]
240 errors aint that bad

**Comment by juan-cortes** [comment]
Come on CI, locally there are no errors, don't ruin my sunday.

File: .changeset/red-spies-shake.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"live-mobile": minor
+---
+
+Migration from JavaScript to TypeScript for LLM
```
Comment (Justkant) [inline .changeset/red-spies-shake.md:5]: I still don't think we need to bump the version of mobile for a refactor

File: apps/ledger-live-mobile/.eslintrc.js [hunk 6]
```diff
@@ -5,15 +5,22 @@ module.exports = {
     "airbnb",
     "prettier",
```
Comment (Justkant) [inline apps/ledger-live-mobile/.eslintrc.js:6]: Should we update prettier to use the same proper config ? https://github.com/prettier/eslint-plugin-prettier#recommended-configuration
```suggestion
    "plugin:prettier/recommended",
```

File: apps/ledger-live-mobile/.eslintrc.js [hunk 7]
```diff
@@ -5,15 +5,22 @@ module.exports = {
...
     "plugin:json/recommended",
+    "plugin:@typescript-eslint/recommended",
```
Comment (Justkant) [inline apps/ledger-live-mobile/.eslintrc.js:8]: We can add the recommended rules for the import plugin: https://github.com/import-js/eslint-plugin-import#typescript
```suggestion
    "plugin:@typescript-eslint/recommended",
    "plugin:import/recommended",
    "plugin:import/typescript",
```

We can probably also add the typed rules for typescript: https://typescript-eslint.io/docs/linting/typed-linting/

File: apps/ledger-live-mobile/.eslintrc.js [hunk 8]
```diff
@@ -5,15 +5,22 @@ module.exports = {
...
       node: {
         extensions: [".js", ".android.js", ".ios.js", ".ts", ".tsx"],
       },
+      typescript: {
+        alwaysTryTypes: true,
+      },
     },
   },
-  plugins: ["prettier", "detox"],
+  plugins: ["prettier", "detox", "@typescript-eslint", "import"],
```
Comment (Justkant) [inline apps/ledger-live-mobile/.eslintrc.js:23]: This should not be needed when using the recommended rules config for `@typescript-eslint` and `import` plugins (and even prettier if we use the correct config):
- https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin#usage
- https://github.com/import-js/eslint-plugin-import#typescript
- https://github.com/prettier/eslint-plugin-prettier#recommended-configuration
```suggestion
  plugins: ["detox"],
```

**Review by Justkant — COMMENTED** [review]
Awesome work, thanks for pushing that forward, you're a beast !!
I think we can improve a little bit the eslint config

**Pull request merged**